src/coreclr/System.Private.CoreLib/src/System/Attribute.CoreCLR.cs

```diff
@@ -301,7 +301,7 @@ private static Attribute[] InternalParamGetCustomAttributes(ParameterInfo param,
                 count = 0;
                 for (int i = 0; i < objAttr.Length; i++)
                 {
-                    if (objAttr[i] is object attr)
+                    if (objAttr[i] is { } attr)
```

Comment:
This is not readable. In the initial version of pattern matching before `is not null`, `is { }` is rejected by many projects.

Reply:
We cannot use `is not null` with _[declaration pattern](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/patterns#declaration-and-type-patterns)_, i.e. `if (objAttr[i] is not null attr)` is invalid C#.

Reply:
```suggestion
                    if (objAttr[i] is not null and var attr)
```

Reply:
Yes, it is unfortunate that there are number of more or less cryptic ways one can express a null check in modern C#. The choice between them is a matter of personal preference. My personal preference is the old fashioned:
```csharp
object? attr = objAttr[i];
if (attr != null)
```
I believe that it is the best balance of readable and succinct. (Except for the few types that overload operator==. It is fine to use to use more verbose `is not null` in those cases for performance reasons.)